lib/capybara/queries/text_query.rb

```diff
@@ -14,11 +14,12 @@ def initialize(*args)
       end
 
       def resolve_for(node)
-        @actual_text = Capybara::Helpers.normalize_whitespace(node.text(@type))
+        @node = node
+        @actual_text = text(node, @type)
         @count = @actual_text.scan(@search_regexp).size
       end
 
-      def failure_message
+      def failure_message(check_invisible = true)
```

Comment:
failure_message and negative_failure_message is an API required by RSpec - I don't think adding method parameters is a good idea in case RSpec every decides to update their API and pass some.  I think it would be better to factor out the message generation pieces into private methods that can then be called from failure_message and negative_failure_message